src/com/amazon/ion/impl/lite/IonSequenceLite.java

```diff
@@ -429,7 +430,7 @@ void writeBodyTo(IonWriter writer, SymbolTableProvider symbolTableProvider)
      *
      * Structural modifications from the sublist itself are allowed.
      */
-    private class SubListView implements List<IonValue> {
+    private class SubListView extends AbstractList<IonValue> {
```

Comment:
Given that we now extend `AbstractList` can we prune down the former implementation (stuff that `AbstractList` brings to the table)?

Reply:
I considered that but didn't look at it too deeply.  Revisit this--there might be some stuff that can be removed.

Reply:
So, as it would turn out we're supposed to throw `ConcurrentModificationException` any time the `SubListView` is accessed and the parent list has been modified.  This means that all of `List<T>` needs to be implemented even if all it does is check if the parent is modified and then delegate to `AbstractList<T>`'s implementation.  I was able to replace the hand-written implementation with `AbstractList<T>`'s in this manner with only about 4 methods.  For the others, either `AbstractList<T>'s` method implementation throws `UnsupportedOperationException` or the override has to interact with the parent `IonSequence` in some way.

I also found a few methods that need to decorate the super Implementation as well...